app/javascript/src/NewService/components/FormTemplate.jsx

```diff
@@ -0,0 +1,116 @@
+// @flow
+
+import React from 'react'
+import {CSRFToken} from 'utilities/utils'
+import type {FormTemplateProps} from 'NewService/types'
+
+type ServiceDiscoveryFormTemplateProps = {
+  hiddenServiceDiscoveryInput: {
+    value: string,
+    name: string,
+    id: string
+  },
+  serviceDiscoverySelect: {
+    [string]: {
+        htmlFor: string,
+        label: string,
+        name: string,
+        id: string
+    }
+  }
+}
+
+const serviceDiscoveryProps: ServiceDiscoveryFormTemplateProps = {
+  hiddenServiceDiscoveryInput: {
+    value: 'discover',
+    name: 'service[source]',
+    id: 'service_source'
+  },
+  serviceDiscoverySelect: {
+    project: {
+      htmlFor: 'namespace',
+      label: 'Namespace',
+      name: 'service[namespace]',
+      id: 'service_namespace'
+    },
+    service: {
+      htmlFor: 'service_name',
+      label: 'Name',
+      name: 'service[name]',
+      id: 'service_name'
+    }
+  }
+}
+
+const renderLabel = (project: {htmlFor: string, label: string}, required: boolean = false) => <label htmlFor={project.htmlFor}>{project.label}
+  { required && <abbr title="required">*</abbr> }
+</label>
+
+const renderSelectOptions = (options: string[]) => options.map(option => <option key={option} value={option}>{option}</option>)
+
+const renderSelect = (project: {name: string, id: string}, props: FormTemplateProps) => <select required="required" name={project.name} id={project.id} onChange={props.onChangeNamespaces}
+  data-selected-value={project.id === 'service_namespace' ? props.selectedNamespace : ''}>
+  {props.namespaces && renderSelectOptions(props.namespaces)}
+</select>
+
+const ErrorMessage = ({error}) => <p style={{color: 'red'}}>{ error.errorMsg && `Sorry, your request has failed with the error: ${error.errorMsg}`}</p>
+
+const renderServiceDiscoveriItems = (props, serviceDiscoveryProps) => {
```

Comment:
For these fn is not necessary to call them `render...` since are just returning components (or fragments). It's a JSX component that, eventually, will be rendered to the dom... but the fn is not in charge of that. This applies to the rest of the "render..." fns.